### PR TITLE
Added grouping.json to bucket 95% of rules.

### PR DESCRIPTION
**Pull request by indexzero**
@olegpesok @mikesherov As per the discussion [on the mailing list](https://groups.google.com/forum/#!topic/jscs-dev/krUsy4TH2JI)

### CODE REVIEW

**Comment by markelog** [comment]
So propose of that list is to use it at jscs.info site? Right? Are you planning to submit supporting PR for the `jscs-dev.github.io` branch?

**Comment by indexzero** [comment]
@markelog yup. Working on that, but the purpose of this PR is to get feedback for the bucketing of the rules into groups themselves.

**Comment by indexzero** [comment]
e.g. might be useful to have a script in this repo that ensures that all rules are in a group?

File: grouping.json [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    "requireParenthesesAroundIIFE",
+    "disallowImplicitTypeConversion",
+    "disallowMultipleLineStrings",
+    "validateQuoteMarks",
+    "disallowYodaConditions",
+    "requireYodaConditions",
+    "disallowNodeTypes",
+    "disallowNotOperatorsInConditionals",
+    "requireDollarBeforejQueryAssignment",
+    "requireNumericLiterals",
```
Comment (hzoo) [inline grouping.json]: Small thing but I added http://jscs.info/rule/requireNumericLiterals.html as a es6 rule since it involves binary/octal literals

**Comment by indexzero** [comment]
@hzoo fixed. I think I should add a test to make sure there is 100% inclusion of all rules. Will try to get to that this weekend.

**Comment by markelog** [comment]
Still interested?

**Comment by indexzero** [comment]
@markelog yup. Just a lot going on. Did some work on this tonight. Look for a branch soon.

**Comment by indexzero** [comment]
@markelog this should be good to go on this side. Now that the rules are all here and verified I can start work on the `jscs-dev.github.io` side.

File: test/specs/grouping.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+var assert = require('assert');
+var fs = require('fs');
+var path = require('path');
+var groups = require('../../grouping');
+
+var rulesDir = path.join(__dirname, '..', '..', 'lib', 'rules');
+
+describe('rule grouping', function () {
+    var optionNames;
```
Comment (hzoo) [inline test/specs/grouping.js]: jshint

`test/specs/grouping.js: line 9, col 9, 'optionNames' is defined but never used.`

File: test/specs/grouping.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    var lookup = Object.keys(groups).reduce(function (acc, group) {
+        groups[group].forEach(function (rule) {
+            acc[rule] = group;
+        });
+
+        return acc;
+    }, {});
+
+    describe.only('exists for every rule', function () {
```
Comment (hzoo) [inline test/specs/grouping.js]: can remove .only

File: test/specs/grouping.js (hunk 2 above)
Comment (indexzero) [inline test/specs/grouping.js]: Will fix, y'all should turn on autofix on your own codebase :smile_cat:

**Comment by indexzero** [comment]
@hzoo comments addressed. Please review.

**Comment by hzoo** [comment]
Looks good - I guess all new rules will have to be added to a group to pass the test!

**Comment by indexzero** [comment]
@hzoo that's the idea. So we can stay up to date as new rules come in.

**Pull request closed without merging**